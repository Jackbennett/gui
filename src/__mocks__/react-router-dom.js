import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
