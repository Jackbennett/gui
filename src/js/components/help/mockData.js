export const helpProps = {
  demoArtifactLink: 'https://dgsbl4vditpls.cloudfront.net/mender-demo-artifact.mender',
  docsVersion: undefined,
  isHosted: false,
  isEnterprise: false,
  menderVersion: 'master',
  menderArtifactVersion: 'master'
};
