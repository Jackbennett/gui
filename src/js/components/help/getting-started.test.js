import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import GettingStarted from './getting-started';
import { helpProps } from './mockData';
import { undefineds } from '../../../../tests/mockData';

describe('GettingStarted Component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <GettingStarted {...helpProps} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
