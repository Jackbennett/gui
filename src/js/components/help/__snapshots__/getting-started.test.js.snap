// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GettingStarted Component renders correctly 1`] = `
<div>
  <h2>
    Get Started
  </h2>
  <p>
    Your devices first need to have the Mender client running on them in order to connect to the server. We support several ways of integrating your device with Mender and connecting it to the Mender server.
  </p>
  <p>
    For a detailed introduction to the different approaches to do this, please refer to our
     
    <a
      href="https://docs.mender.io/get-started"
      rel="noopener noreferrer"
      target="_blank"
    >
      documentation
    </a>
    .
  </p>
</div>
`;
